app/src/actions/entityActions.js

```diff
@@ -0,0 +1,33 @@
+import * as types from './actionTypes';
+import ApiService from "../services/ApiService";
+
+export const setRootEntity = uid => ({
+  type: types.SET_ROOT_ENTITY,
+  uid
+});
+
+export const addEntityWatch = uid => ({
+  type: types.ADD_ENTITY_WATCH,
+  uid
+});
+
+export const fetchEntity = uid => {
+  return dispatch => {
+    return ApiService.getEntity(uid)
+      //TODO:DM - can this hard-coded idx be cleaned up?
+      .then(json => dispatch(receiveEntity(json.objects[0])));
+  };
+};
+
+export const fetchEntities = uids => {
+  return dispatch => {
+    return Promise.all(uids.map(uid =>
```

Comment:
question: is `Promise` needed here?

Reply:
no, I believe not... thanks for asking

at one point I had had a `.then(...)` chaining off of the `Promise.all()`method, which is a method that allows you to combine multiple promises into one, but am no longer using the follow-up clause

i'll confirm that the code works correctly without the extra layer of wrapping all promises returned by the `dispatch()` calls into one that is returned from the thunk func